BraveWallet/Crypto/Portfolio/AddCustomAssetView.swift

```diff
@@ -17,34 +17,58 @@ struct AddCustomAssetView: View {
   @State private var decimalsInput = ""
   @State private var showError = false
   
-  var userAssetStore: UserAssetsStore
-  
   var body: some View {
     NavigationView {
       Form {
         Section(
           header: WalletListHeaderView(title: Text(Strings.Wallet.tokenName))
         ) {
           TextField(Strings.Wallet.enterTokenName, text: $nameInput)
+            .disabled(userAssetStore.isSearchingToken)
         }
         .listRowBackground(Color(.secondaryBraveGroupedBackground))
         Section(
           header: WalletListHeaderView(title: Text(Strings.Wallet.tokenContractAddress))
         ) {
           TextField(Strings.Wallet.enterContractAddress, text: $addressInput)
+            .onChange(of: addressInput) { newValue in
+              if !newValue.isEmpty, newValue.isETHAddress {
+                userAssetStore.tokenInfo(by: newValue) { token in
+                  if let token = token, !token.isErc721 {
+                    if nameInput.isEmpty {
+                      nameInput = token.name
+                    }
+                    if symbolInput.isEmpty {
+                      symbolInput = token.symbol
+                    }
+                    if decimalsInput.isEmpty {
+                      decimalsInput = "\(token.decimals)"
+                    }
+                  }
+                }
+              }
+            }
+            .disabled(userAssetStore.isSearchingToken)
         }
         .listRowBackground(Color(.secondaryBraveGroupedBackground))
         Section(
           header: WalletListHeaderView(title: Text(Strings.Wallet.tokenSymbol))
         ) {
           TextField(Strings.Wallet.enterTokenSymbol, text: $symbolInput)
+            .disabled(userAssetStore.isSearchingToken)
         }
         .listRowBackground(Color(.secondaryBraveGroupedBackground))
         Section(
-          header: WalletListHeaderView(title: Text(Strings.Wallet.decimalsPrecision))
+          header: WalletListHeaderView(title: Text(Strings.Wallet.decimalsPrecision)),
+          footer: Group {
+            ProgressView()
+              .frame(maxWidth: .infinity)
+              .opacity(userAssetStore.isSearchingToken ? 1 : 0)
+          }
```

Comment:
This may be invisible if a user has a keyboard up, maybe we should move it into the address row itself?

Reply:
Example: 

```swift
HStack {
  TextField(NumberFormatter().string(from: NSNumber(value: 0)) ?? "0", text: $decimalsInput)
             .keyboardType(.numberPad)
  if userAssetStore.isSearchingToken {
    ProgressView()
  }
}
```

Reply:
You're introducing some empty space here as well if you're keeping the ProgressView in the footer and making it 0 opacity